### PR TITLE
dot notation support for tags

### PR DESCRIPTION
**Pull request by chunminglu**
when I try to generate html page with some library, it throws an error complaining name of tag is undefined. The reason is that I'm using a dot notation for tags which doesn't have a name property. I think this feature request is reasonable since sometimes you may use multiple libraries which has same component definitions. A dot notation can serve as "namespace" to differentiate components. So I try to add this functionality into this parser.

btw thanks for this beautiful component! It helps a lot for my job to create a dynamic html page from a jsx template!

### CODE REVIEW

File: source/components/JsxParser.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Parser } from 'acorn-jsx'
 import React, { Component, Fragment } from 'react'
+import { get } from 'lodash'
```
Comment (TroyAlford) [inline source/components/JsxParser.js]: :) I won't accept any pull request that introduces a dependency on `lodash`. Though I love the lib, it's huge and bloated, and will massively increase the file-size of `react-jsx-parser`. Can you please refactor to eliminate this dependency?

File: webpack.config.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ const plugins = []
 if (PRODUCTION) {
   plugins.push(
     new webpack.DefinePlugin({ 'process.env.NODE_ENV': JSON.stringify(ENVIRONMENT) }),
-    new webpack.optimize.ModuleConcatenationPlugin(),
```
Comment (TroyAlford) [inline webpack.config.js]: Please put the trailing `,` back in. :) It's required by the linting rules set up for this lib.

**Comment by TroyAlford** [comment]
Overall, thank you for a nice PR! And for the nice compliments. :) I'm glad the component is helping out with your real-world use-cases! If you try my suggestion, above, and it doesn't meet your needs - I'm totally happy to re-review without the `lodash` dependency. :)

I love, love, love user-contributed features like this. :)

File: source/components/JsxParser.js (hunk 1 above)
Comment (chunminglu) [reply source/components/JsxParser.js]: sure! I thought importing partial of the library will only pack functions related to it, but it seems I'm wrong. I'll write it as a seperate helper function and remove reference to lodash.

File: webpack.config.js (hunk 2 above)
Comment (chunminglu) [reply webpack.config.js]: o sure :)

**Comment by chunminglu** [comment]
I update the code according to your suggestions. Thank you for your advice!

**Comment by chunminglu** [comment]
I try to add this dot notation feature is because I'm trying to use this component with some external UI component libraries, and there are two difficulties if I spread the object and directly register components into this parser component. One is that for external UI component libraries, they may defined a component under another component, like "Select" and "Select.Options", so spreading only the root level of library will not help. The other difficulty is that There may be two different UI component libraries that has same name for components. like "Input" from material UI library and "Input" from ant design library. with dot notation, all these two difficulties can be resolved. so I think it is a good supplement for this wonderful component.

**Comment by TroyAlford** [comment]
That makes a bunch of sense. Thanks for clarifying your need. :) I'll pull your branch down and take a look as soon as I can. Given that CI is not passing, I want to hand-verify that everything works properly before I merge anything.

**Comment by TroyAlford** [comment]
Merged and published just now as 1.5.0 :)

**Comment by chunminglu** [comment]
cool! thanks sooo much!

**Pull request merged**